### PR TITLE
Update CMake minimum required version to 3.0.2 (closes #2621)

### LINKED ISSUE #2621: Compiling hpx master fails with /usr/bin/ld: final link failed: Bad value
Using this cmake command with hpx master

cmake  \
-DBOOST_ROOT=/RQusagers/p108903/opt/boost/1.64/ \
-DHWLOC_INCLUDE_DIR=/RQusagers/p108903/opt/hwloc/include  \
-DHWLOC_LIBRARY=/RQusagers/p108903/opt/hwloc/lib/libhwloc.so  \
-DTCMALLOC_LIBRARY=/RQusagers/p108903/opt/gperftools/lib/libtcmalloc_minimal.so  \
-DTCMALLOC_INCLUDE_DIR=/RQusagers/p108903/opt/gperftools/include  \
..

delivers

/usr/bin/ld: CMakeFiles/hpx.dir/performance_counters/registry.cpp.o: relocation R_X86_64_PC32 against `_ZTVN3hpx20performance_counters6server19arithmetics_counterISt10multipliesIdEEE' can not be used when making a shared object; recompile with -fPIC
<diehlpk_work> /usr/bin/ld: final link failed: Bad value

Compiler:

-bash-3.2$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/RQAutofs/users/p108903/opt/gcc/6.3/bin/../libexec/gcc/x86_64-pc-linux-gnu/6.3.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: ./configure --prefix=/RQusagers/p108903/opt/gcc/6.3 -enable-languages=c,fortran,c++ --disable-multilib
Thread model: posix
gcc version 6.3.0 (GCC)

Plattform:

Linux cottos.rqchp.qc.ca 2.6.18-128.el5 #1 SMP Wed Jan 21 10:41:14 EST 2009 x86_64 x86_64 x86_64 GNU/Linux

### PR DESCRIPTION
**Pull request by diehlpk**
This fixes #2621

### CODE REVIEW

**Comment by hkaiser** [comment]
We might want to change the docs as well (https://github.com/STEllAR-GROUP/hpx/blob/master/docs/manual/build_system/prerequisites.qbk#L63 and https://github.com/STEllAR-GROUP/hpx/blob/master/docs/manual/build_system/prerequisites.qbk#L92).

**Comment by diehlpk** [comment]
Updated the documentation.

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,8 +9,8 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-# We require at least CMake V2.8.12
-cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+# We require at least CMake V3.0.2
+cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)
 
 # explicitly set certain policies
 cmake_policy(VERSION 2.8.12)
```
Comment (hkaiser) [inline CMakeLists.txt]: Shouldn't this be changed to 3.0.2 as well?

Comment (diehlpk) [inline CMakeLists.txt]: Sure, will do that now

Comment (hkaiser) [inline CMakeLists.txt]: I'm not saying this _has_ to be changed. I'm asking whether this is necessary... To answer this you'd need to understand what the `cmake_policy(<version>)` macro actually does.

**Comment by hkaiser** [comment]
From what I could read from the docs, using `cmake_policy(VERSION 3.0.2)` in our code should be fine.

**Pull request merged**